### ISSUE TITLE
@blueprintjs/popover2 should declare react-dom as peerDependency

### ISSUE THREAD

**Issue by Grmiade** [palantir/blueprint#5392]
#### Environment

- __Package version(s)__: @blueprintjs/select@4.4.1 & @blueprintjs/popover2@1.4.1

#### Actual behavior

When we use `@blueprintjs/select@4.4.1` with PnP and the new `@blueprintjs/popover2` dependency, we have this error:

```
Cannot find module 'react-dom' from '/root/project/.yarn/__virtual__/@blueprintjs-core-virtual-78dfba3b5a/0/cache/@blueprintjs-core-npm-4.5.1-c4d80a3e45-1c874c7267.zip/node_modules/@blueprintjs/core/lib/cjs/components/context-menu'

    Require stack:
      /root/project/.yarn/__virtual__/@blueprintjs-core-virtual-78dfba3b5a/0/cache/@blueprintjs-core-npm-4.5.1-c4d80a3e45-1c874c7267.zip/node_modules/@blueprintjs/core/lib/cjs/components/context-menu/contextMenu.js
      /root/project/.yarn/__virtual__/@blueprintjs-core-virtual-78dfba3b5a/0/cache/@blueprintjs-core-npm-4.5.1-c4d80a3e45-1c874c7267.zip/node_modules/@blueprintjs/core/lib/cjs/components/index.js
      /root/project/.yarn/__virtual__/@blueprintjs-core-virtual-78dfba3b5a/0/cache/@blueprintjs-core-npm-4.5.1-c4d80a3e45-1c874c7267.zip/node_modules/@blueprintjs/core/lib/cjs/index.js
      /root/project/.yarn/__virtual__/@blueprintjs-popover2-virtual-1e372e8967/0/cache/@blueprintjs-popover2-npm-1.4.1-a69bf38b51-e8024b750b.zip/node_modules/@blueprintjs/popover2/lib/cjs/classes.js
      /root/project/.yarn/__virtual__/@blueprintjs-popover2-virtual-1e372e8967/0/cache/@blueprintjs-popover2-npm-1.4.1-a69bf38b51-e8024b750b.zip/node_modules/@blueprintjs/popover2/lib/cjs/index.js
      /root/project/.yarn/__virtual__/@blueprintjs-select-virtual-672fa868f1/0/cache/@blueprintjs-select-npm-4.4.1-4eddfc7574-3232c04882.zip/node_modules/@blueprintjs/select/lib/cjs/components/multi-select/multiSelect2.js
      /root/project/.yarn/__virtual__/@blueprintjs-select-virtual-672fa868f1/0/cache/@blueprintjs-select-npm-4.4.1-4eddfc7574-3232c04882.zip/node_modules/@blueprintjs/select/lib/cjs/components/index.js
      /root/project/.yarn/__virtual__/@blueprintjs-select-virtual-672fa868f1/0/cache/@blueprintjs-select-npm-4.4.1-4eddfc7574-3232c04882.zip/node_modules/@blueprintjs/select/lib/cjs/index.js
```

#### Expected behavior

We should not have errors.

#### Possible solution

TLDR; `@blueprintjs/popover2` needs `@blueprintjs/core` which has `react-dom` as peerDependency.
So `@blueprintjs/popover2` should also declare `react-dom` as peerDependency in order to not breaking the chain.

When we depend of a package with a peer dependency we have to declare it as dependency.
Here `@blueprintjs/popover2` depends of `@blueprintjs/core` with `react-dom` as peer dependency.
But `@blueprintjs/popover2` doesn't really need `react-dom` because it doesn't use it. So instead of declare it as dependency we should declare it as peer dependency, in order to indicate to the packages which use `@blueprintjs/popover2` they should install `react-dom`, not for `@blueprintjs/popover2` itself but for `@blueprintjs/core`, it's what I called the "chain".
PnP being more strict with this kind of stuff.
